app/src/main/java/com/realityexpander/tasky/agenda_feature/data/common/workers/SyncAgendaWorker.kt

```diff
@@ -0,0 +1,95 @@
+//   & download any new items for the current day.
+@HiltWorker
+class SyncAgendaWorker @AssistedInject constructor(
+    @Assisted val context: Context,
+    @Assisted val workerParams: WorkerParameters,
+    val agendaRepository: IAgendaRepository
+): CoroutineWorker(context, workerParams) {
+
+    override suspend fun doWork(): Result {
+
+        Log.d("SyncAgendaWorker.doWork()", "attemptedRuns: ${workerParams.runAttemptCount}")
+        showNotification(createNotification())
+
+        // Push up local changes to remote
+        val resultSyncAgenda = agendaRepository.syncAgenda()
+        if(resultSyncAgenda is ResultUiText.Success) {
+
+            // Fetch the latest remote changes for today
+            val resultUpdateLocalAgenda = agendaRepository.updateLocalAgendaDayFromRemote(
+                ZonedDateTime.now().truncatedTo(ChronoUnit.DAYS)
+            )
+
+            delay(3000) // prevent flashing notification
+            clearNotification()
+
+            return when (resultUpdateLocalAgenda) {
+                is ResultUiText.Success -> Result.success()
+                is ResultUiText.Error -> Result.failure()
+            }
+        }
+        clearNotification()
+
+        return Result.failure()
+    }
+
+    companion object {
+        const val WORKER_NAME = "SYNC_AGENDA_WORKER"
+        const val NOTIFICATION_ID = 100002
+        const val NOTIFICATION_CHANNEL_ID = NOTIFICATION_SYNC_WORKER_CHANNEL_ID
+    }
+
+    init {
+        val channel = NotificationChannel(
+            NOTIFICATION_CHANNEL_ID,
+            WORKER_NAME,
+            NotificationManager.IMPORTANCE_LOW,
+        )
+        val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        notificationManager.createNotificationChannel(channel)
+    }
+
+    private fun showNotification(notification: Notification) {
+        val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        notificationManager.notify(NOTIFICATION_ID, notification)
+    }
+
+    private fun clearNotification() {
+        val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        notificationManager.cancel(NOTIFICATION_ID)
+    }
+
+    private fun createNotification(): Notification {
+        return NotificationCompat.Builder(context, NOTIFICATION_CHANNEL_ID)
+            .setContentTitle(context.getString(R.string.agenda_sync_notification_title))
+            .setContentText(context.getString(R.string.agenda_sync_uploading_items_text))
+            .setSmallIcon(R.drawable.ic_notification_sync_upload_foreground)
+            .setColor(ResourcesCompat.getColor(context.resources, R.color.tasky_green, null))
+            .setLargeIcon(ResourcesCompat.getDrawable(context.resources, R.drawable.tasky_logo_for_splash, null)?.toBitmap(100,100))
+            .setAutoCancel(true)
+            .build()
+    }
+}
```

Comment:
There's a lot of duplicate code with the other worker, it could make sense to extract that in a separate class (also to follow SRP)

Reply:
OK Will refactor to a worker notification class